File: rb/glicko2_test.rb
#!/usr/bin/env ruby

require 'glicko2'
require 'test/unit'

class TestGlicko2 < Test::Unit::TestCase

	def test_simple
		a = Glicko2.new(1500.0, 200.0, 0.06)
		b = Glicko2.new(1400.0,  30.0, 0.06)
		c = Glicko2.new(1550.0, 100.0, 0.06)
		d = Glicko2.new(1700.0, 300.0, 0.06)
		
		a.add_win(b)
		a.add_loss(c)
		a.add_result(d,0.0)
		
		assert_equal(1500.0, a.rating)
		assert_equal(200.0, a.deviation)
		
		a.update()
		
		assert_in_delta(1464.05,a.rating,0.01)
		assert_in_delta(151.516,a.deviation,0.01)
	end
end

File: go/glicko2_test.go
package glicko2

import (
	"testing"
	"math"
)

func inRange(want float64, have float64, delta float64) bool {
	return math.Abs(want-have) <= delta
}

func TestGlicko2(t *testing.T) {
	a := NewGlicko2(1500.0, 200.0, 0.06)
	b := NewGlicko2(1400.0,  30.0, 0.06)
	c := NewGlicko2(1550.0, 100.0, 0.06)
	d := NewGlicko2(1700.0, 300.0, 0.06)

	a.AddWin(b)
	a.AddLoss(c)
	a.AddLoss(d)

	if a.Rating() != 1500.0 {
		t.Error()
	}
	if a.Deviation() != 200.0 {
		t.Error()
	}

	a.Update()

	if !inRange(a.Rating(),1464.05,0.01) {
		t.Error()
	}
	if !inRange(a.Deviation(),151.516,0.01) {
		t.Error()
	}

	a.Update()

	if !inRange(a.Rating(),1464.05,0.01) {
		t.Error()
	}
	if !inRange(a.Deviation(),151.875,0.01) {
		t.Error()
	}

}
